### PR TITLE
compose with generator-travis

### PR DESCRIPTION
**Pull request by iamstarkov**
I found `composeWith` feature as awesome one and think that if yeoman community will have smaller generators then, everybody will benefit from it. For example with this pull-request, `generator-node` will no longer need to handle travis config on its own. What do you think?

### CODE REVIEW

**Comment by SBoudrias** [comment]
LGTM!

I just want to take a look at the travis generator before actually merging. I'll try to get back to you before the weekend.

File: generators/app/index.js [hunk 1]
```diff
@@ -188,8 +188,8 @@ module.exports = generators.Base.extend({
 
   default: function () {
     if (this.options.travis) {
-      this.composeWith('node:travis', {}, {
-        local: require.resolve('../travis')
+      this.composeWith('travis', {}, {
+         local: require.resolve('generator-travis/generators/app')
```
Comment (arthurvr) [inline generators/app/index.js]: Don't change the indentation here please :)

**Comment by arthurvr** [comment]
LGTM.  Thanks!

Although it works for our use case, I think the generator-travis misses a bunch of things. It would be cool if it had options like what node versions to use, if people even want node at all, custom commands, ... Without those things it's useless for a bunch of generators out there.

**Comment by SBoudrias** [comment]
I have the same question as @arthurvr here, what is the plan for the future of this generator?

**Comment by iamstarkov** [comment]
> what is the plan for the future of this generator?

It is really good question. I spend few days on thinking how to keep balance between single responsibility and extensibility.

Initial idea was to have easy way to get travis config with up to date node versions, like `yo travis`.  Main improvement I can see is to extend existing travis config, (right now it is overriding every time).

I can easily support all the other fields, but I feel like it will blur purpose of this generator. If someone want to have other scripts in travis config, maybe its the job for generator-coveralls or smth. So this generator will only handle node versions and keep already existing fields as before, while other generators will be able to make their job. What do you think?

PS. the reason why I add nodejs as default env, because nodejs community is almost the only yeoman consumer.

PS 2. This Semver upgrade model seems good enough:

1. Patch versions will be updated when generator is about to get small code fixes, or going to remove old node versions.
2. Minor version will be updated, when generator will get new node versions in the list.
3. Major versions will be updated, when generator will get breaking change for generator-consumers

**Comment by iamstarkov** [comment]
Second thought is to support all custom fields, because then everybody still can use up-to-date node versions and just extend travis config. So right now roadmap looks like this: 1. extend travis config 2. support all custom fields. What do you think?

**Comment by iamstarkov** [comment]
Delivered with generator-travis 1.1

**Comment by iamstarkov** [comment]
only one problem I found, fields in result travis config is not sorted properly.  For example:

```
writing: function () {
  this.composeWith('travis', { options: {
    config: { after_script: 'npm run coveralls' }
  }}, {
    local: require.resolve('generator-travis/generators/app')
  });
},
```
Will be:

```
after_script: 'npm run coveralls'
language: node_js
node_js:
  - stable
  - iojs
  - '0.12'
  - '0.10'
```

Is there any way to sorted JSON.stringify?

**Comment by iamstarkov** [comment]
Yay, I [imlemented sorted Yaml config](https://github.com/iamstarkov/generator-travis/commit/7e19122e91aa69225c49f4b328e4569247a3bc10). Take a look pls.

**Comment by iamstarkov** [comment]
I’m also thinking to remove old node versions from the list, because whole point to get "up-to-date", it will be true by default with possibility to turn off in options or with flag.

So what do you think about pull-request?

**Comment by iamstarkov** [comment]
@SBoudrias @arthurvr looking for your feedback =)

**Comment by SBoudrias** [comment]
> I’m also thinking to add feature of removing old node versions from the list, because whole point is to get "up-to-date" travis config, it will be true by default with possibility to turn off in options or with flag.

IMO, I'd settle on a default config (0.10, 0.12, stable) and allow an option to manually specify versions. I'd be simpler.

It's kind of important currently supporting 0.10 to 4.1 because apt-get default is still 0.10, so a lot of people still have it installed.

--------------

For this CR, I'm okay to merge. But fix the indentation issue and update the readme too as there's references to the travis sub-generator.

**Comment by iamstarkov** [comment]
* fixed indentation /cc @arthurvr 
* update generator-node readme /cc @SBoudrias 
* add ability to specify nodejs versions in the `config` options /cc @SBoudrias 

Also I wanted the more clear way to understand how to add or remove versions in the list, so I found [nodejs/LTS](https://github.com/nodejs/LTS/#readme) plan and ended up [relevant section](https://github.com/iamstarkov/generator-travis#nodejs-versions-in-the-config) in the generator-travis’s readme. What do you think?

**Comment by SBoudrias** [comment]
Cool, LGTM.

Please squash your commits and we'll merge.

**Comment by iamstarkov** [comment]
done

**Comment by SBoudrias** [comment]
Here it goes! Thanks a ton

**Comment by hemanth** [comment]
:+1:

**Pull request merged**